### ISSUE TITLE
Weight vs Measurement in scale() / conditioning example

### ISSUE THREAD

**Issue by rseymour** [pyro-ppl/pyro#1808]
### Issue Description
Conditioning on measurement guarantees all of the conditioned scale functions in chapter 2 of the docs return `9.5` in the example.  It's more interesting to condition based on a true weight measurement, and I think it shows more of what you're trying to show, a tightening of the distribution.  The docs could keep an example of conditioning on `measurement`, essentially hardcoding the return, but it would be useful to switch in a conditioned on `weight` example.

```
conditioned_scale = pyro.condition(scale, data={"weight": 9.8})
print conditioned_scale(8.6)
Out:  tensor(9.9817)

conditioned_scale = pyro.condition(scale, data={"measurement": 9.8})
print conditioned_scale(3)
Out:  9.8
```

(note the differing return types as well.) I can do this doc fix if one of y'all thinks it is appropriate., I just want to know for sure that I'm not missing something. I think the equations and measurement | guess, weight all point to conditioning on weight as the clearer example.  Thanks again.

**Comment by eb8680** [comment]
> It's more interesting to condition based on a true weight measurement

Hmm, not sure I follow this reasoning.  Why would it be more interesting to condition on a true weight?  The point of the scale example is that we don't have access to true weights because the scale is unreliable, but we can still use a model of the measurement noise plus an initial uncertain guess to obtain an uncertain estimate of the true weight.  Note also that `pyro.condition` fixes values of random variables but does not perform inference, so anything returned from `conditioned_scale` is not statistically meaningful, no matter which variables you condition on or return.

**Comment by rseymour** [comment]
Well, to stick in the example world, it's common with scales to have test samples of known mass.  In this case put 10g test mass on, you guess the scale will say 11g and so forth.  But by "more interesting" conditioning on the last returned value is the same as saying 

```
def scale(guess):
    # ignore all sampling and inputs
    return 9.5
```

Still learning here so this threw me for a loop causing me to come in and open this ticket.  If you condition on something 'inside' of the chain of distribution calculations you can see how the shift happens in the random output.  If you do it as the return value you need to then place the scale() inside of another setup to see.  I think I was just expecting the conditioning to be something more 'complex' than a simple hardcode of the return value, even though conditioning is truly just that.  I'm going to close this. Thanks for your enlightenment, I would state in plain english that all 3 given variations all do the same thing, which is return 9.5 regardless of input.
